### PR TITLE
[nrf noup] Spake2+ alignments regarding the newest Oberon version

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Aligned Spake2+ algorithm to the Oberon PSA core v1.2.1.1

### CODE REVIEW

File: src/crypto/PSASpake2p.h [hunk 1]
```diff
@@ -155,6 +155,7 @@ class PSASpake2p_P256_SHA256_HKDF_HMAC
     CHIP_ERROR GetKeys(SessionKeystore & keystore, HkdfKeyHandle & key);
 
 private:
+    psa_pake_cipher_suite_t mCipherSuite = PSA_PAKE_CIPHER_SUITE_INIT;
```
Comment (Damian-Nordic) [inline src/crypto/PSASpake2p.h:158]: Does it have to live as long as the operation? If not, shouldn't it be allocated on stack?

**Review by Damian-Nordic — APPROVED** [review]

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 2]
```diff
@@ -284,58 +284,39 @@ CHIP_ERROR PsaKdf::Init(const ByteSpan & secret, const ByteSpan & salt, const By
...
 
-        if (salt.size() > 0)
-        {
-            status = psa_key_derivation_input_bytes(&mOperation, PSA_KEY_DERIVATION_INPUT_SALT, salt.data(), salt.size());
-            VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
-        }
-
-        status = psa_key_derivation_input_key(&mOperation, PSA_KEY_DERIVATION_INPUT_SECRET, hkdfKey.mKeyId);
-        VerifyOrReturnError(status == PSA_SUCCESS, CHIP_ERROR_INTERNAL);
+    status = psa_key_derivation_setup(&mOperation, PSA_ALG_HKDF(PSA_ALG_SHA_256));
```
Comment (kkasperczyk-no) [inline src/crypto/CHIPCryptoPALPSA.cpp:299]: Nit: you could assign it one line above :)

**Review by kkasperczyk-no — APPROVED** [review]

File: src/crypto/PSASpake2p.h (hunk 1 above)
Comment (Damian-Nordic) [reply src/crypto/PSASpake2p.h]: Right.

**Comment by ArekBalysNordic** [comment]
DNM before Chip plans do not pass: https://github.com/nrfconnect/sdk-nrf/pull/14136

**Review by frkv — APPROVED** [review]
LGTM

**Pull request merged**